src/ispypsa/translator/temporal_filters.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime, timedelta
+
+import pandas as pd
+
+from ispypsa.config.validators import ModelConfig
+
+
+def apply_snapshot_filter(time_series_data, snapshot):
+    """Filters a timeseries pandas DataFrame based using the datetime values in
+     the snapshot index.
+
+    Examples:
+
+    >>> datetime_index = pd.date_range('2020-01-01', '2020-01-03', freq='H')
+    >>> time_series_data = pd.DataFrame({'Datetime': datetime_index, 'Value': range(len(datetime_index))})
+    >>> snapshot = pd.DataFrame(index=datetime_index[::12])  # Every 12 hours
+    >>> apply_snapshot_filter(time_series_data, snapshot)
+                  Datetime  Value
+    0  2020-01-01 00:00:00      0
+    12 2020-01-01 12:00:00     12
+    24 2020-01-02 00:00:00     24
+    36 2020-01-02 12:00:00     36
+    48 2020-01-03 00:00:00     48
+
+    """
+    return time_series_data[time_series_data["Datetime"].isin(snapshot.index)]
+
+
+def filter_snapshot(config: ModelConfig, snapshot: pd.DataFrame):
```

Comment:
Could feed in AggregationConfig if you change structure as suggested in config comment

Reply:
Made this change.